src/utils/components/EnvironmentEditor/EnvironmentForm.tsx

```diff
@@ -35,69 +40,81 @@ const EnvironmentForm: React.FC<EnvironmentFormProps> = ({ vm, onEditChange, upd
     onEnvironmentAdd,
     onEnvironmentChange,
     onEnvironmentRemove,
-    onReload,
-    onSave,
     edited,
     error: formError,
-    setError,
-  } = useEnvironments(vm, updateVM, onEditChange);
+    setError: setFormError,
+  } = useEnvironments(temporaryVM, vm, setTemporaryVM, onEditChange);
+
+  useEffect(() => {
+    setTemporaryVM(vm);
+    setFormError(null);
+  }, [setFormError, setTemporaryVM, vm]);
 
   const environmentNamesSelected = environments.map((env) => env.name);
 
+  const onReload = useCallback(() => {
+    setTemporaryVM((draftVM) => {
+      draftVM.spec = vm.spec;
+    });
+  }, [vm.spec, setTemporaryVM]);
+
   if (!loaded) return <EnvironmentFormSkeleton />;
 
   return (
-    <Form>
-      <EnvironmentFormTitle />
-      {environments.length !== 0 && (
-        <div className="row pairs-list__heading">
-          <div className="col-xs-5 text-secondary text-uppercase" id="environment-name-header">
-            {t('config map / secret / service account')}
+    <SidebarEditor<V1VirtualMachine> resource={temporaryVM} onChange={setTemporaryVM}>
```

Comment:
Why adding the `onChange` method in the SidebarEditor `props`? 
In the previous cases, the resource in the sidebar was already edited with modals. So in the modal, the user clicks submit, and the updated resource is given in input to the sidebareditor component. Also, the user could edit the resource from the editor and save it. The edited resource was provided in the form with the children as a function to give the 'live' experience.

In the Environments tab, there is no modal but just a form. To give the `live` editor experience I just removed the editor Save button for this case in order to have only one submit button. What does this mean? That we need to have one shared editable resource (`temporaryVM`) in both ways simultaneously.